### ISSUE TITLE
Coppermine Photo Galleries - Blocked images

### ISSUE THREAD

**Issue by THEtomaso** [wolfbeast/lunarblocklist#7]
### Issue:
Blocked images in some Coppermine Photo Galleries.

### Elaboration:
Coppermine Photo Galleries are widely used by celebrity fansites.
Some of these sites's photo galleries have "ads" sections, which contains images of the celebrity participating in various ad campaigns.

### Example (gallery images are blocked):
https://elizabeth-olsen.com/photos/thumbnails.php?album=967

### Problem rule:
`/ads/*$image`

### Possible fix:
`@@/photos/albums/*/ads$image`
(A bit too generic perhaps?)

--

### System info:
* OS/version: Windows 8.1 Pro (x64)
* Browser/version: Pale Moon v28.10.0 (x64)
* Adblock Extension/version: uBlock Origin v1.16.4.23b2 (XUL)

**Comment by wolfbeast** [comment]
The suggested exception rule is too generic.

**Comment by THEtomaso** [comment]
Yeah, I tend to agree.
Perhaps something like this then, for now?:
`@@/photos/albums/*/ads$image,domain=elizabeth-olsen.com`
..then you can add more domains, when needed.

**Comment by wolfbeast** [comment]
No, I'm not going to make updates for every single one of the domains of "widely used celebrity fansites"; that would be a prayer without end.

**Comment by THEtomaso** [comment]
> that would be a prayer without end.

Indeed, this issue affects many sites.
This is why a generic fix would is sorely needed!
I first discovered this issue, after being tasked with this thing, a couple of years ago:
https://github.com/uBlockOrigin/uAssets/issues/3401/
Some sites needs a whitelisting for `/advertisements/*` (found in **EasyList**), while others apparently needs it for `/ads/*$image` (now found in your filter).
I reported a few of these issues to begin with, but soon didn't bother with them anymore, since I couldn't really come up with a decent generic fix.

**Comment by wolfbeast** [comment]
I think there might be a way to deal with this specifically if the fan sites are indeed all using the same software that will generate the same kinds of overview pages.
Can you investigate if there is a common pattern for embedded filenames, not just paths?
Can you investigate if viewing full images would also be affected by the filter or not?

**Comment by THEtomaso** [comment]
> Can you investigate if there is a common pattern for embedded filenames, not just paths?

Unfortunately, the webmasters seems to have total freedom when it comes to filenames.
Although, a large portion of them seems to use what appears to be Coppermine's default file naming scheme, which is simply `001.jpg`, `002.jpg`, and so on (also used in the example that I posted).

--

> Can you investigate if viewing full images would also be affected by the filter or not?

I just tested with a site that I had reported for **EasyList** here:
https://github.com/easylist/easylist/issues/2436/
When overriding EasyList's current whitelisting in uBO (using `/advertisements/*$important`), the thumbnails becomes blocked again here:
https://ellefanningfan.net/photos/thumbnails.php?album=1217
..but the full screen images are still shown (even if I add the `$image` variable to the blocking rule, since `$document` is required for direct file links):
https://ellefanningfan.net/photos/albums/upload/advertisements/2018/miufall/006.jpg
I suspect that this might have different outcomes, depending on which adblocker you're using though.
